src/runtime/task/mod.rs

```diff
@@ -58,11 +58,20 @@ pub(crate) struct Task {
     name: Option<String>,
 
     local_storage: StorageMap,
+    pub(super) span: tracing::Span,
 }
 
 impl Task {
     /// Create a task from a continuation
-    fn new<F>(f: F, stack_size: usize, id: TaskId, name: Option<String>, clock: VectorClock) -> Self
+    fn new<F>(
+        f: F,
+        stack_size: usize,
+        id: TaskId,
+        name: Option<String>,
+        clock: VectorClock,
+        parent_span: &tracing::Span,
```

Comment:
`Span`s are just handles, so I'd just do `tracing::Span` here and everywhere else